rfc/rfc000-package-server.md

```diff
@@ -1,19 +1,91 @@
+# Package Server
+
 ---
+
 RFC: 000
-Author: Peter Bourgon <peter@bourgon.org>
+
+Author: Matt Heath <matt@mattheath.com>
+
 Status: Draft
----
 
-# package server
+---
 
 ## Motivation
 
-http://peter.bourgon.org/go-kit/#package-server
+[http://peter.bourgon.org/go-kit/#package-server](http://peter.bourgon.org/go-kit/#package-server)
+
+Package server is probably the biggest and most important component of the toolkit. Ideally, we should be able to write our services as implementations of normal, nominal Go interfaces, and delegate integration with the environment to the server package. The package should encode and enforce conventions for server-side concerns, like health checks, system-wide request tracing, connection management, backpressure and throttling, and so on. For each of those topics, it should provide interfaces for different, pluggable strategies. It should integrate with service discovery, and work equally well over multiple transports. Considerable prior art exists in the form of Finagle, Karyon (Netflixs application service library), and likely many more.
 
 ## Scope
 
-To be defined.
+### Endpoints
+
+*   An endpoint is defined as a handler interface which receives a request and returns a response or an error to the client.
```

Comment:
@tsenart that's a bit of a tautology (esp since the project is truly in its infancy and we are literally discussing the defined goals as we speak!), but so be it. RPC is certainly the most important thing to support. But I can think of numerous examples where streaming responses were absolutely essential to proper system behavior. Happy to elaborate if that's necessary.